packages/core/analytics/core-analytics-browser-internal/src/track_viewport_size.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { debounceTime, fromEvent, map, merge, of, shareReplay } from 'rxjs';
+import type { AnalyticsClient, RootSchema } from '@kbn/analytics-client';
+
+export interface ViewportSize {
+  viewport_width: number;
+  viewport_height: number;
+}
+
+const schema: RootSchema<ViewportSize> = {
+  viewport_width: {
+    type: 'long',
+    _meta: { description: 'The value seen as the CSS viewport @media (width)' },
+  },
+  viewport_height: {
+    type: 'long',
+    _meta: { description: 'The value seen as the CSS viewport @media (height)' },
+  },
+};
+
+// window or document?
+// According to MDN, it only emits on `window`: https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event
+const resize$ = fromEvent(window, 'resize').pipe(
```

Comment:
I wonder if we shouldn't put the `resize$` declaration/definition within `trackViewportSize` (given the function should be called only one anyway). It feels more natural to avoid that kind of static variables. 

OTOH the perf impact is insignificant so, idk